torch/_functorch/_aot_autograd/functional_utils.py

```diff
@@ -406,16 +406,16 @@ def assert_functional_graph(fx_g: torch.fx.Graph) -> int:
             if n.target in [
                 torch.ops.aten.copy_.default,
                 torch.ops.aten.set_.source_Tensor,
+                torch.ops.fsdp.set_.default,
             ]:
                 suffix = True
-                # Can only copy_/set_ into an input, and can only do so once
+                # Can only copy_/set_ into an input
                 # this is mostly a hack to avoid failing XLA tests.
                 # See https://github.com/pytorch/pytorch/pull/122434#issuecomment-2101012113
                 if "set_buffer_donor_" not in str(n.args[0]):
                     assert (
                         n.args[0] in placeholders
                     ), f"n={str(n)}, n.args[0]={str(n.args[0])}, placeholders={str(placeholders)}, graph={str(fx_g)}"
-                    placeholders.remove(n.args[0])
```

Comment:
oh yeah removing this assert... makes sense (we can now end up with multiple mutations on the same graph input.